File: labor1/app/src/main/java/com/example/labor1/MainActivity.java
package com.example.labor1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {
    // TODO: implement onCallBack

    private static int callFromProfile = 0;
    private static int callFromRegister = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            String caller = bundle.getString("caller");
            if (Objects.equals(caller, "profile")) {
                callFromProfile++;
            } else if (Objects.equals(caller, "register")) {
                callFromRegister++;
            }
        }
        ((TextView) this.findViewById(R.id.profileCallCounter)).setText(String.valueOf(callFromProfile));
        ((TextView) this.findViewById(R.id.registerCallCounter)).setText(String.valueOf(callFromRegister));
    }

    public void callRegister(View view) {
        Intent registerActivity = new Intent(this, registerActivity.class)
                .putExtra("caller", "login");
        startActivity(registerActivity);
    }

    public void callProfile(View view) {
        Intent profileActivity = new Intent(this, ProfileActivity.class)
                .putExtra("firstName", ((EditText) this.findViewById(R.id.firstNameText)).getText().toString())
                .putExtra("lastName", ((EditText) this.findViewById(R.id.lastNameText)).getText().toString())
                .putExtra("caller", "login");
        startActivity(profileActivity);
    }
}

File: labor1/app/src/main/java/com/example/labor1/MainActivity.kt
package com.example.labor1;

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class MainActivity : AppCompatActivity() {

    companion object {
        @JvmStatic
        var callFromProfile = 0
        @JvmStatic
        var callFromRegister = 0
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bundle = intent.extras
        if (bundle != null) {
            val caller = bundle.getString("caller")
            if (Objects.equals(caller, "profile")) {
                callFromProfile++
            } else if (Objects.equals(caller, "register")) {
                callFromRegister++
            }
        }
        findViewById<TextView>(R.id.profileCallCounter).text = callFromProfile.toString()
        findViewById<TextView>(R.id.registerCallCounter).text = callFromRegister.toString()
    }

    fun callRegister(view: View) {
        val registerActivity = Intent(this, RegisterActivity::class.java)
                .putExtra("caller", "login")
        startActivity(registerActivity)
    }

    fun callProfile(view: View) {
        val profileActivity = Intent(this, ProfileActivity::class.java)
                .putExtra("firstName", findViewById<EditText>(R.id.firstNameText).text.toString())
                .putExtra("lastName", findViewById<EditText>(R.id.lastNameText).text.toString())
                .putExtra("caller", "login");
        startActivity(profileActivity);
    }
}
